src/include/btree_inline.h

```diff
@@ -1793,41 +1793,20 @@ static inline bool
 __wt_page_evict_retry(WT_SESSION_IMPL *session, WT_PAGE *page)
 {
     WT_PAGE_MODIFY *mod;
-    WT_TXN_GLOBAL *txn_global;
```

Comment:
I don't think it's OK to remove the checks this function does related to transaction state. This part of the change doesn't seem to be described in the ticket. Is it necessary to get the test case passing?

I went and looked and the checks were added as part of WT-3745 and WT-3763. Those were done to improve performance with MongoDB, and as far as I can tell those workloads are still relevant.

Reply:
The checks on the oldest id and the pinned timestamp is effectively the same as the check not to evict the pages with newer updates. After I remove `(modified && page->modify->update_txn >= conn->txn_global.last_running)`, the check !__wt_page_evict_retry(session, page)) is still blocking the page being queued for eviction.